### PR TITLE
Remove feature flag to show "Get started" button always

### LINKED ISSUE #7510: WooCommerce Account Creation
### Description 
Create a new screen that given an email and password creates a new WordPress account

### Tasks

- [x] Add new "Get Started" button to prologue under feature flag that opens new Sign Up screen:  #7534 
- [x] Create new screen with given UI: #7534 
- [x] Add API requests for happy path: [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2544) and #7565
- [x] Error handling: #7584
- [x] Tracking #7673
- [x] Trigger store creation flow after successful signup  #7662 
- [x] Handle "Log in" CTA navigation to log in with WP.com credentials #7662 

### Designs 

Screenshot 2022-10-11 at 15 05 29

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #7510 

### Description
- Remove feature flag to display "Get started" button always.
- ~~Add the final changes to the login prologue UI~~
- Add unit tests
- Handle edge case where user add the `wordPress` keyword as part of their emails. This will make fetching `username` suggestions provide invalid user names containing `wordPress` which is a blacklited word for usernames. 
- ~~Add release notes~~ (No release notes added as we don't want to advertise this change) 

### Testing instructions
Out of the changes added in this PR the only thing to test is to try to create an account using an email with thisformat: 
`whatever+wordPress@email.com`
The account should be created successfully.

### CODE REVIEW

**Comment by hichamboushaba** [comment]
Hey @JorgeMucientes 👋, do you think the changes on this PR will be conflicting with what @hafizrahman is working on the other PR #7634? if so, then maybe you should coordinate together 🙂.

**Comment by JorgeMucientes** [comment]
Yes @hichamboushaba, once we have the final design for the `control` variant of the prologue screen we'll need to coordinate as is it very likely there are going to be conflicts. @hafizrahman and I already talked about it last Friday. Whoever merges things first the other we'll have to deal with conflicts 👍🏼

**Comment by hafizrahman** [comment]
I'm working on the latest design of Prologue (both for control and simplified login variants) in #7634

The Prologue changes in this PR does not seem to be too significant, so it shouldn't be too complex to deal with the conflicts.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/signup/SignUpCredentialsValidator.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package com.woocommerce.android.ui.login.signup
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/signup/SignUpCredentialsValidator.kt:1]: This logic is unchanged. I simply extracted it out of the `SignUpRepository`

**Review by 0nko — APPROVED** [review]
Looking good! 💯

**Comment by 0nko** [comment]
We're good to merge once the conflicts are resolved.

**Comment by JorgeMucientes** [comment]
Hey @0nko thanks for reviewing! 
I resolved conflicts and additionally I replaced the feature flag `STORE_CREATION_WEBVIEW_FLOW` with new feature flag `NATIVE_STORE_CREATION_FLOW` that will be used to kick off the native flow after account creation for `debug` builds. You can check the changes in this [commit](https://github.com/woocommerce/woocommerce-android/pull/7676/commits/77cdc78b9eb06559647bab39f9aa31823245997f)

**Comment by 0nko** [comment]
Awesome, I was going to do this but you beat me to it! :)
